File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        for(char ch : s){
            if(ch == '{' || ch == '(' || ch == '['){
                st.push(ch);
            }else{
                if(st.empty()) return false;

                char tp  = st.top();
                st.pop();
                if((ch == ')'  && tp != '(') || (ch == '}' && tp != '{') || (ch == ']' && tp != '['))return false;
            }
            

        }
        return st.empty();
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {

    let stack = [];

    for (let c of s) {

        // Opening bracket
        if (c === '(' || c === '{' || c === '[') {
            stack.push(c);
        }

        // Closing bracket
        else {

            if (stack.length === 0) {
                return false;
            }

            let top = stack[stack.length - 1];

            if (c === ')' && top !== '(') {
                return false;
            }

            if (c === '}' && top !== '{') {
                return false;
            }

            if (c === ']' && top !== '[') {
                return false;
            }

            stack.pop();
        }
    }

    return stack.length === 0;
};